### PR TITLE
Add comprehensive tests for error handlers

### LINKED ISSUE #86: Create YAML Workflow Examples Collection
**Description:**
Create a collection of 5 common workflow examples in YAML format for different use cases to help users understand Hive's workflow capabilities.

**Acceptance Criteria:**
- [ ] Create `examples/workflows/` directory
- [ ] Add 5 workflow examples: data processing, API integration, file operations, scheduled tasks, error handling
- [ ] Include detailed comments explaining each workflow step
- [ ] Add a README in examples/workflows/ explaining how to use them

**Getting Started:**
- Review existing workflow documentation
- Check if `examples/` directory exists
- Study Hive's YAML workflow syntax

**Resources:**
- Hive documentation
- Existing configuration examples

**Part of Initiative:** https://github.com/namastexlabs/automagik-roadmap/issues/74

### PR DESCRIPTION
**Pull request by ManishhDev**
This file contains comprehensive tests for error handling utilities, including the ModelProviderErrorHandler and handle_model_errors decorator. It tests various error types and edge cases to ensure proper handling and logging.

## 🔗 Linked Issue/Wish (Optional but Recommended)

**Linked Issues:**
Closes #86
**Wish Path:**

---

## 📝 Summary

---

## 🔧 Changes Implemented

1. **Component 1**
   - Change A
   - Change B

2. **Component 2**
   - Change C

---

## ✅ Testing

- [ ] Tests added/updated
- [ ] All tests passing locally
- [ ] Manual testing completed

---

## 💥 Breaking Changes

- [ ] Yes (describe migration path below)
- [x] No

---

## 📸 Screenshots/Evidence (Optional)

<details>
<summary>Before/After</summary>

</details>

---

## 📚 Additional Context (Optional)

---

### CODE REVIEW

**Review by vasconceloscezar — COMMENTED** [review]
...

File: tests/lib/utils/test_error_handlers.py [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+        
+        @handle_model_errors(agent_id="test_agent")
+        async def async_func_success():
+            return {"result": "success"}
+        
+        result = await async_func_success()
+        
+        assert result["result"] == "success"
+
+    def test_sync_function_model_provider_error_handling(self):
```
Comment (vasconceloscezar) [inline tests/lib/utils/test_error_handlers.py]: ❌ **Problem:** Same issue - decorator checks `str(type(e))` not error message.

✅ **Fix:** Add mock class:
```python
class ClientError(Exception):
    pass
```

Change to:
```python
raise ClientError("rate limit")
```

File: tests/lib/utils/test_error_handlers.py [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+        
+        with pytest.raises(ValueError) as exc_info:
+            sync_func_with_non_model_error()
+        
+        assert "not a model provider error" in str(exc_info.value)
+
+    @pytest.mark.asyncio
+    async def test_decorator_without_agent_id(self):
+        """Test decorator works without explicit agent_id."""
+
```
Comment (vasconceloscezar) [inline tests/lib/utils/test_error_handlers.py]: ❌ **Problem:** Using plain Exception won't match decorator's type check.

✅ **Fix:** Add mock class:
```python
class APIError(Exception):
    pass
```

Change to:
```python
raise APIError("something went wrong")
```

File: tests/lib/utils/test_error_handlers.py [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+        """Test detection of various model provider error types."""
+        error_types = [
+            "ModelProviderError",
+            "ClientError",
+            "ServerError",
+            "APIError",
+            "AuthenticationError",
+            "google.genai.errors",
+            "GoogleAIError",
+            "GenerativeAIError",
```
Comment (vasconceloscezar) [inline tests/lib/utils/test_error_handlers.py]: ❌ **Problem:** Loop creates Exception objects, not the named error types.

✅ **Fix:** Create dynamic exception types:
```python
for error_type in error_types:
    @handle_model_errors(agent_id="test_agent")
    async def async_func():
        error_class = type(error_type, (Exception,), {})
        raise error_class("test error")
```

File: tests/lib/utils/test_error_handlers.py [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+        assert result["kwarg1"] == "value3"
+
+    def test_decorator_with_sync_function_arguments(self):
+        """Test decorator works with sync functions that have arguments."""
+        
+        @handle_model_errors(agent_id="test_agent")
+        def sync_func_with_args(arg1, arg2, kwarg1=None):
+            if kwarg1 == "error":
+                raise Exception("APIError: test error")
+            return {"arg1": arg1, "arg2": arg2}
```
Comment (vasconceloscezar) [inline tests/lib/utils/test_error_handlers.py]: ❌ **Problem:** Plain Exception won't be caught by decorator.

✅ **Fix:** Use mock APIError class:
```python
if kwarg1 == "error":
    raise APIError("test error")
```
